src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs

```diff
@@ -100,7 +100,14 @@ public async Task<List<CloudApplication>> GetCompatibleApplications(List<Recomme
             foreach (var app in allDeployedApplications)
             {
                 if (recommendations.Any(rec => string.Equals(rec.Recipe.Id, app.RecipeId, StringComparison.Ordinal)))
+                {
                     compatibleApplications.Add(app);
+                }
+                // check for persisted custom deployment recommendations
+                else if (recommendations.Any(rec => string.Equals(rec.Recipe.BaseRecipeId, app.RecipeId, StringComparison.Ordinal)))
```

Comment:
This 2 check process of for finding a recommendation for a given target recipe id is repeated here and on lines 271 and 275 of the DeployCommand. This seems like an easy thing in the future to forget doing 2 check process if we have another use case. Can we put this in a utility method?

Reply:
Fixed